astropy/coordinates/inf_dist.py

```diff
@@ -0,0 +1,16 @@
+from contextlib import contextmanager
+
+_inf_dist_enabled = False  # Global flag
```

Comment:
I don't know how to feel about undocumented global variable. We have config and we have science states, why not leverage one of those instead?